debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java

```diff
@@ -62,7 +63,7 @@
     private final Metronome metronome;
     private final Supplier<PreviousContext> loggingContextSupplier;
 
-    private volatile Throwable producerFailure;
+    private volatile ConnectException producerFailure;
```

Comment:
Should not e start using DebeziumException?

Reply:
We'd still have to throw it as KC's `RetriableException` in the end, so I don't think there's much point in changing this atm.